### PR TITLE
Ideas

### PR DESCRIPTION
**Pull request by allen4397**
Hey guys.

Here is the refactor for biggest bust and biggest surprise. I used other methods we had created. I added a comment for a method I thought we could move to the team class. I also added an extra two data files because I worked for an hour to get the test right for these methods but the only way I could get it to work was to make the extra files. Adding games to the original file was really messy and cause a lot of problems, and the mocks wouldn't work because I called the same method twice with two different arguments (apparently that doesn't work for stubs).

Anyway, hopefully this looks a lot better.

Also went ahead and replaced that inject enum with sum we talked about earlier.

Curious what to address next. I guess, Teresa, we will talk tomorrow?

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -235,114 +224,26 @@ def winningest_team
...
-      elsif season_id == game.season && game.type == "R"
-        reg_season << game
-      end
+  def group_games_by_season_type(type, games = @games)
+    games.select do |game|
+      game.type == type
     end
+  end
+
+  def biggest_bust(season_id)
```
Comment (teresa-m-knowles) [inline lib/stat_tracker.rb]: this looks much better

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -235,114 +224,26 @@ def winningest_team
...
+    regular_season = group_games_by_season_type(("R"), games_by_season[season_id])
+    preseason = group_games_by_season_type(("P"), games_by_season[season_id])
```
Comment (teresa-m-knowles) [inline lib/stat_tracker.rb]: @allen4397 Hey Tim, why is the "P" between parenthesis here?

Comment (allen4397) [reply lib/stat_tracker.rb]: Oh, I see what your saying. I think I was just tired and wasn't thinking. Looking at it now, it's a string argument so it should be fine without it. I can take it out.

**Comment by allen4397** [comment]
What is the plan for merging this branch?

**Pull request merged**